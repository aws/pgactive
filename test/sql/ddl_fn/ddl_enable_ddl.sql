SET pgactive.skip_ddl_replication = true;
SELECT pgactive.pgactive_replicate_ddl_command($DDL$ CREATE TABLE public.should_fail ( id integer ) $DDL$);

RESET pgactive.skip_ddl_replication;
SELECT pgactive.pgactive_replicate_ddl_command($DDL$ CREATE TABLE public.create_ok (id integer) $DDL$);

SET pgactive.skip_ddl_replication = true;
SELECT pgactive.pgactive_replicate_ddl_command($DDL$ ALTER TABLE public.create_ok ADD COLUMN alter_should_fail text $DDL$);

RESET pgactive.skip_ddl_replication;
SELECT pgactive.pgactive_replicate_ddl_command($DDL$ DROP TABLE public.create_ok $DDL$);

-- Now for the rest of the DDL tests, presume they're allowed,
-- otherwise they'll get pointlessly verbose.
SELECT pgactive.pgactive_replicate_ddl_command($DDL$ ALTER DATABASE regression RESET pgactive.skip_ddl_replication $DDL$);
SELECT pgactive.pgactive_replicate_ddl_command($DDL$ ALTER DATABASE postgres RESET pgactive.skip_ddl_replication $DDL$);
