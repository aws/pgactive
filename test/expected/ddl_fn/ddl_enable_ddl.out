SET pgactive.skip_ddl_replication = true;
SELECT pgactive.pgactive_replicate_ddl_command($DDL$ CREATE TABLE public.should_fail ( id integer ) $DDL$);
ERROR:  pgactive_replicate_ddl_command execution attempt rejected by configuration
DETAIL:  pgactive.skip_ddl_replication is true.
HINT:  See the 'DDL replication' chapter of the documentation.
RESET pgactive.skip_ddl_replication;
SELECT pgactive.pgactive_replicate_ddl_command($DDL$ CREATE TABLE public.create_ok (id integer) $DDL$);
 pgactive_replicate_ddl_command 
---------------------------
 
(1 row)

SET pgactive.skip_ddl_replication = true;
SELECT pgactive.pgactive_replicate_ddl_command($DDL$ ALTER TABLE public.create_ok ADD COLUMN alter_should_fail text $DDL$);
ERROR:  pgactive_replicate_ddl_command execution attempt rejected by configuration
DETAIL:  pgactive.skip_ddl_replication is true.
HINT:  See the 'DDL replication' chapter of the documentation.
RESET pgactive.skip_ddl_replication;
SELECT pgactive.pgactive_replicate_ddl_command($DDL$ DROP TABLE public.create_ok $DDL$);
 pgactive_replicate_ddl_command 
---------------------------
 
(1 row)

-- Now for the rest of the DDL tests, presume they're allowed,
-- otherwise they'll get pointlessly verbose.
SELECT pgactive.pgactive_replicate_ddl_command($DDL$ ALTER DATABASE regression RESET pgactive.skip_ddl_replication $DDL$);
 pgactive_replicate_ddl_command 
---------------------------
 
(1 row)

SELECT pgactive.pgactive_replicate_ddl_command($DDL$ ALTER DATABASE postgres RESET pgactive.skip_ddl_replication $DDL$);
 pgactive_replicate_ddl_command 
---------------------------
 
(1 row)

